File: tags/REL_5_4/Plugins/Kingbotk/AWB Plugin/WikiProjects/WPAlbums.vb
Friend NotInheritable Class WPAlbums
    Inherits PluginBase

    Private Const PluginName As String = "WikiProject Albums"
    Const Prefix As String = "Albums"

    Friend Sub New()
        MyBase.New("Album|Albums|WP Albums|WPAlbums") ' Specify alternate names only

        Dim params(-1) As TemplateParameters

        OurSettingsControl = New GenericWithWorkgroups("WikiProject Albums", Prefix, True, params)
    End Sub

    ' Settings:
    Private ReadOnly OurTab As New TabPage("Albums")
    Private WithEvents OurSettingsControl As GenericWithWorkgroups

    Protected Friend Overrides ReadOnly Property PluginShortName() As String
        Get
            Return "Albums"
        End Get
    End Property
    Protected Overrides ReadOnly Property PreferredTemplateName() As String
        Get
            Return PluginName
        End Get
    End Property

    Protected Overrides Sub ImportanceParameter(ByVal Importance As Importance)
        Template.NewOrReplaceTemplateParm("importance", Importance.ToString, article, False, False)
    End Sub
    Protected Friend Overrides ReadOnly Property GenericSettings() As IGenericSettings
        Get
            Return OurSettingsControl
        End Get
    End Property
    Protected Overrides ReadOnly Property CategoryTalkClassParm() As String
        Get
            Return "NA"
        End Get
    End Property
    Protected Overrides ReadOnly Property TemplateTalkClassParm() As String
        Get
            Return "NA"
        End Get
    End Property
    Protected Overrides ReadOnly Property FileTalkClassParm() As String
        Get
            Return "File"
        End Get
    End Property

    ' Initialisation:
    Protected Friend Overrides Sub Initialise()
        OurMenuItem = New ToolStripMenuItem("Albums Plugin")
        InitialiseBase() ' must set menu item object first
        OurTab.UseVisualStyleBackColor = True
        OurTab.Controls.Add(OurSettingsControl)
    End Sub

    ' Article processing:
    Protected Overrides Function SkipIfContains() As Boolean
        Return False
    End Function
    Protected Overrides Sub ProcessArticleFinish()
        StubClass()
    End Sub
    Protected Overrides Function TemplateFound() As Boolean
        ' Nothing to do here
    End Function

    Protected Overrides Function WriteTemplateHeader() As String
        Return "{{album" & WriteOutParameterToHeader("class") & _
           WriteOutParameterToHeader("importance")
    End Function

    'User interface:
    Protected Overrides Sub ShowHideOurObjects(ByVal Visible As Boolean)
        PluginManager.ShowHidePluginTab(OurTab, Visible)
    End Sub

    ' XML settings:
    Protected Friend Overrides Sub ReadXML(ByVal Reader As XmlTextReader)
        Dim blnNewVal As Boolean = PluginManager.XMLReadBoolean(Reader, Prefix & "Enabled", Enabled)
        If Not blnNewVal = Enabled Then Enabled = blnNewVal ' Mustn't set if the same or we get extra tabs
        OurSettingsControl.ReadXML(Reader)
    End Sub
    Protected Friend Overrides Sub Reset()
        OurSettingsControl.Reset()
    End Sub
    Protected Friend Overrides Sub WriteXML(ByVal Writer As XmlTextWriter)
        Writer.WriteAttributeString(Prefix & "Enabled", Enabled.ToString)
        OurSettingsControl.WriteXML(Writer)
    End Sub

    ' Misc:
    Friend Overrides ReadOnly Property HasReqPhotoParam() As Boolean
        Get
            Return False
        End Get
    End Property
    Friend Overrides Sub ReqPhoto()
    End Sub
End Class

File: Plugins/Kingbotk Plugin/WikiProjects/WPAlbums.cs
using AutoWikiBrowser.Plugins.Kingbotk;
using AutoWikiBrowser.Plugins.Kingbotk.Plugins;

using System.Windows.Forms;
using System.Xml;

internal sealed class WPAlbums : PluginBase
{

	private const string PluginName = "WikiProject Albums";

	const string Prefix = "Albums";
	internal WPAlbums() : base("Album|Albums|WP Albums|WPAlbums")
	{
		// Specify alternate names only

		TemplateParameters[] @params = new TemplateParameters[-1 + 1];

		OurSettingsControl = new GenericWithWorkgroups("WikiProject Albums", Prefix, true, @params);
	}

	// Settings:
	private readonly TabPage OurTab = new TabPage("Albums");

	private readonly GenericWithWorkgroups OurSettingsControl;
	protected internal override string PluginShortName {
		get { return "Albums"; }
	}
	protected override string PreferredTemplateName {
		get { return PluginName; }
	}

	protected override void ImportanceParameter(Importance Importance)
	{
		Template.NewOrReplaceTemplateParm("importance", Importance.ToString(), article, false, false);
	}
	protected internal override IGenericSettings GenericSettings {
		get { return OurSettingsControl; }
	}

	// Initialisation:
	protected internal override void Initialise()
	{
		OurMenuItem = new ToolStripMenuItem("Albums Plugin");
		InitialiseBase();
		// must set menu item object first
		OurTab.UseVisualStyleBackColor = true;
		OurTab.Controls.Add(OurSettingsControl);
	}

	// Article processing:
	protected override bool SkipIfContains()
	{
		return false;
	}
	protected override void ProcessArticleFinish()
	{
		StubClass();
	}
	protected override bool TemplateFound()
	{
		// Nothing to do here
	    return false;
	}

	protected override string WriteTemplateHeader()
	{
		return "{{album" + WriteOutParameterToHeader("class") + WriteOutParameterToHeader("importance");
	}

	//User interface:
	protected override void ShowHideOurObjects(bool Visible)
	{
		PluginManager.ShowHidePluginTab(OurTab, Visible);
	}

	// XML settings:
	protected internal override void ReadXML(XmlTextReader Reader)
	{
		Enabled = PluginManager.XMLReadBoolean(Reader, Prefix + "Enabled", Enabled);
		OurSettingsControl.ReadXML(Reader);
	}
	protected internal override void Reset()
	{
		OurSettingsControl.Reset();
	}
	protected internal override void WriteXML(XmlTextWriter Writer)
	{
		Writer.WriteAttributeString(Prefix + "Enabled", Enabled.ToString());
		OurSettingsControl.WriteXML(Writer);
	}

	// Misc:
	internal override bool HasReqPhotoParam {
		get { return false; }
	}
	internal override void ReqPhoto()
	{
	}
}
